cryptography/src/commonMain/kotlin/com/wire/kalium/cryptography/E2EIClient.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.cryptography
+
+typealias JsonRawData = ByteArray
+typealias AcmeAccount = ByteArray
+typealias AcmeOrder = ByteArray
+
+data class AcmeDirectory(
+    var newNonce: String,
+    var newAccount: String,
+    var newOrder: String
+)
+
+data class NewAcmeOrder(
+    var delegate: JsonRawData,
+    var authorizations: List<String>
+)
+
+data class AcmeChallenge(
+    var delegate: JsonRawData,
+    var url: String
+)
+
+data class NewAcmeAuthz(
+    var identifier: String,
+    var wireHttpChallenge: AcmeChallenge?,
+    var wireOidcChallenge: AcmeChallenge?
+)
+
+data class AcmeFinalize(
+    var delegate: JsonRawData,
+    var certificateUrl: String
+)
+
+@Suppress("TooManyFunctions")
+interface E2EIClient {
+    fun directoryResponse(directory: JsonRawData): AcmeDirectory
+    fun newAccountRequest(
+        directory: AcmeDirectory,
+        previousNonce: String
+    ): JsonRawData
+
+    @Suppress("LongParameterList")
+    fun newOrderRequest(
+        displayName: String,
+        domain: String,
+        clientId: String,
+        handle: String,
+        expiryDays: UInt,
+        directory: AcmeDirectory,
+        account: AcmeAccount,
+        previousNonce: String
+    ): JsonRawData
+
+    fun newOrderResponse(order: JsonRawData): NewAcmeOrder
+    fun newAuthzRequest(
+        url: String,
+        account: AcmeAccount,
+        previousNonce: String
+    ): JsonRawData
+
+    fun newAuthzResponse(authz: JsonRawData): NewAcmeAuthz
+
+    @Suppress("LongParameterList")
+    fun createDpopToken(
+        accessTokenUrl: String,
+        userId: String,
+        clientId: ULong,
+        domain: String,
```

Comment:
I checked it, it's only getting string and it's not the same the clientId object that we pass to core-crypto itself. Should be use `CryptoQualifiedClientId` anyway?

Reply:
A client Id is an hex encoded string and it can also be represented as number. That's also how it represented in the protobuf messages. See https://github.com/wireapp/kalium/blob/ee8f3de419953c84f13340f0c4c6220f3d59ee7c/network/src/commonMain/kotlin/com/wire/kalium/network/api/base/authenticated/message/OtrClientIdMapper.kt#L27

So we code move the `decodeHexToLong` to `utils` and implement a `encodeLongAsHex`.

I'm ok with doing this in a separate PR.

Reply:
I'd suggest to keep the clientId as it is now and only introduce a `CryptoQualifiedUserId`

Reply:
The representation for the Client ID seems to be inconsistent in the e2ei library, sometimes it's a string and sometimes it's a `Ulong`. 

Maybe something you give as feedback to Beltram.

Reply:
As you mentioned, I asked Beltram and he will update the library later with ClientID to make a consistency for all clientIds.